ehealthid-cli/src/main/java/com/oviva/ehealthid/cli/KeyGeneratorCommand.java

```diff
@@ -31,20 +46,57 @@ public Integer call() throws Exception {
     var encName = "enc";
 
     logger.atInfo().log("generating signing keys");
-    var federationSigningKeys = generateJwks(KeyUse.SIGNATURE);
+    var federationSigningKeys = generateSigningKey(issuerUri);
 
     logger.atInfo().log("generating encryption keys");
-    var federationIdTokenEncryptionKey = generateJwks(KeyUse.ENCRYPTION);
+    var federationIdTokenEncryptionKey = generateEncryptionKey();
 
-    saveJwks(sigName, federationSigningKeys);
-    saveJwks(encName, federationIdTokenEncryptionKey);
+    saveJwks(sigName + "_" + deriveName(issuerUri), new JWKSet(List.of(federationSigningKeys)));
+    saveJwks(encName + "_" + deriveName(issuerUri), new JWKSet(federationIdTokenEncryptionKey));
 
     return 0;
   }
 
-  private JWKSet generateJwks(KeyUse keyUse) throws JOSEException {
-    var key = new ECKeyGenerator(Curve.P_256).keyUse(keyUse).keyIDFromThumbprint(true).generate();
-    return new JWKSet(key);
+  private String deriveName(URI issuer) {
```

Comment:
can you provide a clearer name? maybe `deriveFdpNameFromHost` or something similar?